Create database not visible in the pg_stat_statements view

kritika-0601 opened the pull request:
Currently Babelfish does not show create database query inside the pg_stat_statements table. After my change now it is visible in the view.
pg_stat_statements displays planning and execution statistics for all SQL statements executed by a server. Some statements like create and drop database was not being tracked. Mainly because subsequent processutility hooks after bbf_processutilty were not being called.
Created a hook for that and called it inside relevant ProcessUtility function.

Link for engine PR: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/114

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
Create database not visible in the pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty were not being
called. Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

kritika-0601 committed:
Merge branch 'BABEL_3_X_DEV' into pg_stat_statements-CreateDb

Commit:
retrigger tests

kritika-0601 committed:
Merge branch 'BABEL_3_X_DEV' into pg_stat_statements-CreateDb

Commit:
Create database not visible in the pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty were not being
called. Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
Create database not visible in the pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty were not being
called. Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
Create database not visible in the pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty were not being
called. Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
retrigger tests

Commit:
retrigger tests

Commit:
Create database not visible in the pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty were not being
called. Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

Commit:
retrigger tests

forestkeeper commented on contrib/babelfishpg_tsql/src/pl_handler.c:
```diff
@@ -3330,22 +3330,15 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
-				return;
-			}
-			break;
+        // case T_DropdbStmt:
```
would you just delete those lines instead of keep them commented

forestkeeper commented on contrib/babelfishpg_tsql/src/hooks.c:
```diff
@@ -294,10 +300,17 @@ InstallExtendedHooks(void)
 	modify_RangeTblFunction_tupdesc_hook = modify_RangeTblFunction_tupdesc;
 
+	prev_CreateDbStmt_hook = CreateDbStmt_hook;
+    CreateDbStmt_hook = pltsql_CreateDbStmt;
```
Space

Commit:
Create database not visible in the pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty were not being
called. Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>

kritika-0601 committed:
Merge branch 'babelfish-for-postgresql:BABEL_3_X_DEV' into pg_stat_statements-CreateDb

forestkeeper commented on contrib/babelfishpg_tsql/src/hooks.c:
```diff
@@ -340,14 +353,33 @@ UninstallExtendedHooks(void)
+{
+	Node	   *parsetree = pstmt->utilityStmt;
+ 	DropdbStmt *stmt = (DropdbStmt *) parsetree;
+    drop_bbf_db(stmt->dbname, stmt->missing_ok, false);
```
tab / space not match

kritika-0601 committed:
Merge branch 'BABEL_3_X_DEV' into pg_stat_statements-CreateDb

Commit:
retrigger tests

Commit:
Create database not visible in the pg_stat_statements view

Subsequent processutility hooks after bbf_processutilty were not being
called. Created a hook for that and called it inside relevant
ProcessUtility function.

Signed-off-by: Kritika <kritikaq@amazon.com>
